javascript/javascript1/week2/flightFullname.js

```diff
@@ -0,0 +1,32 @@
+// function getFullname(firstname, surname){
+//     return firstname + ' ' + surname;
+// }
+
+
+// // Formal fullname
+// function getFullname(firstname, surname, useFormalName) {
+//     return useFormalName ? 'Lord ' + firstname + ' ' + surname : 
+//     firstname + ' ' + surname;
+// }
+
+
+function getFullname(firstname, surname, useFormalName=true, gender) {
+    return firstname === '' || surname === '' ? 'Please provide your firstname and surname.' : 
+    useFormalName && gender === 'male' ? 'Lord ' + firstname + ' ' + surname : 
+    useFormalName && gender === 'female' ? 'Lady ' + firstname + ' ' + surname :
+    firstname + ' ' + surname;
+}
+
+
+const fullname1 = getFullname('Alice', 'White', 'female');
```

Comment:
Does `getFullname('Alice', 'White', 'female')` return what you expect? And if so or not so, why? 🧐

Reply:
But your default value for `useFormalName` is true? 😄 I will just cut to the chase, haha, in this case you are passing the value `'female'` as an argument into `useFormalName`, but since you are not passing a `gender`, it does not try to add Lord or Lady. I hope it makes sense. 🙂